client/signup/steps/clone-destination/index.jsx

```diff
@@ -0,0 +1,185 @@
+/** @format */
+/**
+ * External dependencies
+ */
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { localize } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import StepWrapper from 'signup/step-wrapper';
+import Card from 'components/card';
+import Button from 'components/button';
+import SignupActions from 'lib/signup/actions';
+import FormTextInput from 'components/forms/form-text-input';
+import FormLabel from 'components/forms/form-label';
+import ExternalLink from 'components/external-link';
+
+class CloneDestinationStep extends Component {
+	static propTypes = {
+		flowName: PropTypes.string,
+		goToNextStep: PropTypes.func.isRequired,
+		positionInFlow: PropTypes.number,
+		signupProgress: PropTypes.array,
+		stepName: PropTypes.string,
+		signupDependencies: PropTypes.object,
+	};
+
+	state = {
+		form: {
+			destinationSiteName: '',
+			destinationSiteUrl: '',
+		},
+	};
+
+	handleFieldChange = ( { target: { name, value } } ) =>
+		this.setState( {
+			form: Object.assign( this.state.form, { [ name ]: value } ),
```

Comment:
This mutates state and it's not something that should be done. To avoid this, assign to an empty object:
```js
form: Object.assign( {}, this.state.form, { [ name ]: value } ),
```
You can verify this in console with:
```js
const object1 = { a: 1, b: 2, c: 3 };
const object2 = { a: 4, b: 5, c: 6 };

Object.assign( {}, object1, object2 );
console.log( 'object1 unmodified', object1 );

Object.assign( object1, object2 );
console.log( 'object1 is mutated', object1 );
```